### DISCUSSION TITLE
How to set `defaultValue` for a date?

### DISCUSSION THREAD

**Discussion by Notaduck** [react-hook-form/react-hook-form#5907] [Q&A] [answered]
Hi there.  

I have a `date` field in my form:
```js
<input className={`appearance-none block w-full  text-gray-700 border rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white ${errors?.dob && "border-red-600 border-3"}`} 
type="date"
{...register('dob', { valueAsDate: true})}
/>
```

When it is submitted it is stored in a state component like this:

![Screenshot 2021-07-12 at 13 38 21](https://user-images.githubusercontent.com/9454064/125281370-6ea49e80-e316-11eb-8ea0-9726c77c3ed2.png)

and I am trying to set the default value like this:

```js
    const { register, handleSubmit, watch, formState: { errors }, setValue } = useForm<IRegisterUser>({
        reValidateMode: 'onChange',
        resolver: yupResolver(registerUserSchema),
        defaultValues: {
            dob: formState.dob  || "",
        },
    });
```

However it keeps being empty on initial render even though the state isn't empty.

**Comment by bluebill1049** [comment, marked as answer]
what do you mean for `dates`? you mean multiple dates?

**Reply by Notaduck** [reply]
> what do you mean for `dates`? you mean multiple dates?
One single date, sorry.

**Reply by bluebill1049** [reply]
native input should be kept as `string`

**Reply by bluebill1049** [reply]
you can try to get it work with native input first `2021-07-02' then plug into defaultValues
